### ISSUE TITLE
Classmap wildcards supported or not?

### ISSUE THREAD

**Issue by tbreuss** [composer/getcomposer.org#176]
The current documentation states that wildcards are supported:

> Wildcards (*) are also supported in a classmap paths, and expand to match any directory name:

See: <https://getcomposer.org/doc/04-schema.md#classmap>

If I do so, composer quits with a runtime exception:

~~~
[RuntimeException]                                                                                   
Could not scan for classes inside "folder_with_classes/*" which does not appear to be a file nor a folder 
~~~

I use composer version 1.10.17 on macOS.

**Comment by Seldaek** [comment]
Can you show the output of `find folder_with_classes` ?

**Comment by tbreuss** [comment]
Here is an excerpt of the output.

~~~
folder_with_classes
folder_with_classes/minify
folder_with_classes/minify/JSMin.php
folder_with_classes/minify/Minify_HTML.php
folder_with_classes/minify/Minify_CSS_Compressor.php
folder_with_classes/library
folder_with_classes/library/TcpdfList.php
folder_with_classes/library/global.php
folder_with_classes/library/FpdfList.php
folder_with_classes/library/KeyValueModel.php
etc.
~~~

**Comment by Seldaek** [comment]
Ah sorry I missed your `I use composer version 1.10.17 on macOS.`. This feature is only available from Composer 2.0.0. 

That said, just using `folder_with_classes/` is enough as it will anyway be parsed recursively. The wildcard is only useful in the middle of the path IMO.

**Comment by tbreuss** [comment]
👍 +1
